### PR TITLE
remove ruamel_yaml import/except fallback

### PR DESCRIPTION
**Pull request by dholth**
### Description

We removed `ruamel_yaml` a year ago.

The `conda install required-dependency` errors are odd - they might be a little hard to follow through on...

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda/common/configuration.py [hunk 1]
```diff
@@ -54,14 +54,9 @@
...
-    try:
-        from ruamel_yaml.comments import CommentedMap, CommentedSeq
-        from ruamel_yaml.reader import ReaderError
-        from ruamel_yaml.scanner import ScannerError
-    except ImportError:
-        raise ImportError(
-            "No yaml library available. To proceed, conda install ruamel.yaml"
-        )
+    raise ImportError(
+        "No yaml library available. To proceed, conda install ruamel.yaml"
```
Comment (dholth) [inline conda/common/configuration.py:58]: Should this message go away or be replaced with "reinstall conda with a different environment's conda"?

Comment (jezdez) [reply conda/common/configuration.py:58]: Good question!

**Comment by jezdez** [comment]
@dholth I don't think we need this fallback at all anymore since we're depending on `ruamel.yaml` now and it should just be installed upon upgrade, right?

**Review by jezdez — APPROVED** [review]
